### PR TITLE
[VNext][Alan Coding] Productize repo-worker package and retire reference staging

### PR DESCRIPTION
**Pull request by realmorrisliu**
## Summary
- add the first-party `repo-coding` builtin package and package-local `repo-worker` child root
- move smoke, harness, fixtures, and task names from `coding-reference` to `repo-worker`
- replace the standalone reference coding spec with the steward/worker contract and remove top-level `reference/coding-agent` staging

## Testing
- cargo fmt --all
- cargo test -p alan-runtime skills::capability_view::tests::resolved_capability_view_includes_builtin_packages -- --exact
- cargo test -p alan-runtime skills::registry::tests::load_capability_view_applies_skill_overrides -- --exact
- bash scripts/repo-worker/run_smoke.sh --mode ci
- bash scripts/harness/run_repo_worker_suite.sh --ci-blocking

## Follow-ups
- stack repo-worker capability-hardening slices on top of this productization PR
- keep steward benchmark and evaluation work as follow-up PRs

Part of #10, #282, #283, #285

### CODE REVIEW

File: scripts/harness/run_repo_worker_suite.sh [hunk 1]
```diff
@@ -114,19 +90,9 @@ for fixture_rel in "${fixtures[@]}"; do
     started_at="$(date -u +%Y-%m-%dT%H:%M:%SZ)"
     scenario_start_epoch="$(date +%s)"
     set +e
-    validate_exact_cargo_filters "$scenario_id" "$scenario_cmd" >"$scenario_dir/precheck.log" 2>&1
-    precheck_exit=$?
+    (cd "$repo_root" && bash -lc "$scenario_cmd") >"$scenario_dir/event_trace.log" 2>&1
+    exit_code=$?
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply scripts/harness/run_repo_worker_suite.sh]: Restored the exact-filter precheck in `run_repo_worker_suite.sh` using the same `cargo test ... -- --exact --list` validation pattern as the autonomy and compaction suites. I also narrowed the repo-worker exact-test scenarios to `--lib` so the precheck stays targeted and does not silently pass on zero-match filters.

File: scripts/harness/run_repo_worker_suite.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@ set -euo pipefail
 usage() {
     cat <<'USAGE'
 Usage:
-  scripts/harness/run_coding_reference_suite.sh [--ci-blocking]
+  scripts/harness/run_repo_worker_suite.sh [--ci-blocking]
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply scripts/harness/run_repo_worker_suite.sh:7]: Kept the backward-compatible entrypoint by adding `scripts/harness/run_coding_reference_suite.sh` as a thin wrapper around `run_repo_worker_suite.sh`. That fixes the current workflow/job immediately without forcing the check name or workflow wiring to move in the same PR.
